src/utils/fiat_oracle_types.ts

```diff
@@ -0,0 +1,4 @@
+import type { CoinGeckoTokenId, CoinGeckoVSCurrency } from './coingecko_types';
+
+export type TokenID = CoinGeckoTokenId;
```

Comment:
I'm not sure this is really a valid abstraction if we're just setting the types as equal. This wouldn't let us, say, use these types with a different backend provider. I think you still need a mapping layer from one type to the other. I could see TokenID being, for example, an enum type and their being a factory function in the oracle implementation layer to map from oracle implementation types to oracle interface types.

Reply:
So, we have to choose which currencies to put on our side of the mapping. And to make them different (maybe just using mayus)

Reply:
As discussed during our call, we can reverse the provenance here - let's use Coingecko's values as our own and then assign OUR type into the coingecko type.

Reply:
Here it is the mapping feat 5ba66799e09ee81d006c0662218e968aeb04c9e2